dnn/model/yolov3-tiny.py

```diff
@@ -0,0 +1,248 @@
+
+    x = YoloConvTiny(256, name='yolo_conv_0')(x)
+    output_0 = YoloOutput(256, len(masks[0]), classes, name='yolo_output_0')(x)
+    x = YoloConvTiny(128, name='yolo_conv_1')((x, x_8))
+    output_1 = YoloOutput(128, len(masks[1]), classes, name='yolo_output_1')(x)
+
+    if training:
+        return Model(inputs, (output_0, output_1), name='yolov3_tiny')
+
+    boxes_0 = Lambda(lambda x: yolo_boxes(x, anchors[masks[0]], classes), name='yolo_boxes_0')(output_0)
+    boxes_1 = Lambda(lambda x: yolo_boxes(x, anchors[masks[1]], classes), name='yolo_boxes_1')(output_1)
+
+    outputs = Lambda(lambda x: yolo_nms(x), name='yolo_nms')((boxes_0[:3], boxes_1[:3]))
+    return Model(inputs, outputs, name='yolov3_tiny')
+
+
+def YoloLoss(anchors, classes=FLAGS.class_number, ignore_thresh=0.5):
+    def yolo_loss(y_true, y_pred):
+        # 1. transform all pred outputs
+        # y_pred: (batch_size, grid, grid, anchors, (x, y, w, h, obj, ...cls))
+        pred_box, pred_obj, pred_class, pred_xywh = yolo_boxes(y_pred, anchors, classes)
+        pred_xy = pred_xywh[..., 0:2]
+        pred_wh = pred_xywh[..., 2:4]
+
+        # 2. transform all true outputs
+        # y_true: (batch_size, grid, grid, anchors, (x1, y1, x2, y2, obj, cls))
+        true_box, true_obj, true_class_idx = tf.split(y_true, (4, 1, 1), axis=-1)
+        true_xy = (true_box[..., 0:2] + true_box[..., 2:4]) / 2
+        true_wh = true_box[..., 2:4] - true_box[..., 0:2]
+
+        # give higher weights to small boxes
+        box_loss_scale = 2 - true_wh[..., 0] * true_wh[..., 1]
+
+        # 3. inverting the pred box equations
+        grid_size = tf.shape(y_true)[1]
+        grid = tf.meshgrid(tf.range(grid_size), tf.range(grid_size))
+        grid = tf.expand_dims(tf.stack(grid, axis=-1), axis=2)
+        true_xy = true_xy * tf.cast(grid_size, tf.float32) - tf.cast(grid, tf.float32)
+        true_wh = tf.math.log(true_wh / anchors)
+        true_wh = tf.where(tf.math.is_inf(true_wh), tf.zeros_like(true_wh), true_wh)
+
+        # 4. calculate all masks
+        obj_mask = tf.squeeze(true_obj, -1)
+        # ignore false positive when iou is over threshold
+        best_iou = tf.map_fn(
+            lambda x: tf.reduce_max(broadcast_iou(x[0], tf.boolean_mask(x[1], tf.cast(x[2], tf.bool))), axis=-1),
+            (pred_box, true_box, obj_mask),
+            tf.float32)
+        ignore_mask = tf.cast(best_iou < ignore_thresh, tf.float32)
+
+        # 5. calculate all losses
+        xy_loss = obj_mask * box_loss_scale * tf.reduce_sum(tf.square(true_xy - pred_xy), axis=-1)
+        wh_loss = obj_mask * box_loss_scale * tf.reduce_sum(tf.square(true_wh - pred_wh), axis=-1)
+        obj_loss = binary_crossentropy(true_obj, pred_obj)
+        obj_loss = obj_mask * obj_loss + (1 - obj_mask) * ignore_mask * obj_loss
+        true_class_one_hot = tf.one_hot(tf.cast(true_class_idx[..., 0], tf.int32), classes)
+        class_loss = obj_mask * binary_crossentropy(true_class_one_hot, pred_class)
```

Comment:
I don't think ```binary_crossentropy``` works here if training with multiple classes is allowed.

Reply:
However, original darknet implementation also uses binary_crossentropy.
Also, someone did an experiment to use binary_crossentropy here, and it works.
```
715/Unknown - 129s 181ms/step - loss: 819.5117 - yolo_output_0_loss: 28.9505 - yolo_output_1_loss: 104.6755 - yolo_output_2_loss: 675.1611
Epoch 00001: saving model to checkpoints/yolov3_train_1.tf
715/715 [==============================] - 298s 416ms/step - loss: 819.5117 - yolo_output_0_loss: 28.9505 - 
yolo_output_1_loss: 104.6755 - yolo_output_2_loss: 675.1611 - val_loss: 180.2229 - val_yolo_output_0_loss: 6.5867 - val_yolo_output_1_loss: 18.5627 - val_yolo_output_2_loss: 144.3504 - lr: 1.0000e-04
Epoch 2/50
715/715 [==============================] - ETA: 0s - loss: 110.8124 - yolo_output_0_loss: 5.3276 - yolo_output_1_loss: 12.9779 - yolo_output_2_loss: 81.7900
Epoch 00002: saving model to checkpoints/yolov3_train_2.tf
715/715 [==============================] - 292s 409ms/step - loss: 110.8124 - yolo_output_0_loss: 5.3276 - yolo_output_1_loss: 12.9779 - yolo_output_2_loss: 81.7900 - val_loss: 71.5413 - val_yolo_output_0_loss: 4.7071 - val_yolo_output_1_loss: 9.2439 - val_yolo_output_2_loss: 46.8816 - lr: 1.0000e-04
Epoch 3/50
715/715 [==============================] - ETA: 0s - loss: 54.7577 - yolo_output_0_loss: 3.8612 - yolo_output_1_loss: 7.8642 - yolo_output_2_loss: 32.3329
Epoch 00003: saving model to checkpoints/yolov3_train_3.tf
715/715 [==============================] - 291s 407ms/step - loss: 54.7577 - yolo_output_0_loss: 3.8612 - yolo_output_1_loss: 7.8642 - yolo_output_2_loss: 32.3329 - val_loss: 43.5827 - val_yolo_output_0_loss: 4.1549 - val_yolo_output_1_loss: 6.4818 - val_yolo_output_2_loss: 22.2565 - lr: 1.0000e-04
Epoch 4/50
715/715 [==============================] - ETA: 0s - loss: 36.5493 - yolo_output_0_loss: 3.0293 - yolo_output_1_loss: 5.5990 - yolo_output_2_loss: 17.2411
Epoch 00004: saving model to checkpoints/yolov3_train_4.tf
715/715 [==============================] - 288s 403ms/step - loss: 36.5493 - yolo_output_0_loss: 3.0293 - yolo_output_1_loss: 5.5990 - yolo_output_2_loss: 17.2411 - val_loss: 32.7991 - val_yolo_output_0_loss: 4.2579 - val_yolo_output_1_loss: 5.0273 - val_yolo_output_2_loss: 12.8445 - lr: 1.0000e-04
Epoch 5/50
715/715 [==============================] - ETA: 0s - loss: 27.8374 - yolo_output_0_loss: 2.2460 - yolo_output_1_loss: 4.2862 - yolo_output_2_loss: 10.6454
Epoch 00005: saving model to checkpoints/yolov3_train_5.tf
715/715 [==============================] - 292s 408ms/step - loss: 27.8374 - yolo_output_0_loss: 2.2460 - yolo_output_1_loss: 4.2862 - yolo_output_2_loss: 10.6454 - val_loss: 27.5969 - val_yolo_output_0_loss: 4.6249 - val_yolo_output_1_loss: 4.4424 - val_yolo_output_2_loss: 7.8793 - lr: 1.0000e-04
Epoch 6/50
715/715 [==============================] - ETA: 0s - loss: 22.6884 - yolo_output_0_loss: 1.5510 - yolo_output_1_loss: 3.3140 - yolo_output_2_loss: 7.1820
Epoch 00006: saving model to checkpoints/yolov3_train_6.tf
715/715 [==============================] - 291s 407ms/step - loss: 22.6884 - yolo_output_0_loss: 1.5510 - yolo_output_1_loss: 3.3140 - yolo_output_2_loss: 7.1820 - val_loss: 25.6328 - val_yolo_output_0_loss: 5.0260 - val_yolo_output_1_loss: 4.3830 - val_yolo_output_2_loss: 5.5915 - lr: 1.0000e-04
Epoch 7/50
715/715 [==============================] - ETA: 0s - loss: 19.3426 - yolo_output_0_loss: 1.1537 - yolo_output_1_loss: 2.4202 - yolo_output_2_loss: 5.1452
Epoch 00007: saving model to checkpoints/yolov3_train_7.tf
715/715 [==============================] - 290s 406ms/step - loss: 19.3426 - yolo_output_0_loss: 1.1537 - yolo_output_1_loss: 2.4202 - yolo_output_2_loss: 5.1452 - val_loss: 24.6496 - val_yolo_output_0_loss: 5.3167 - val_yolo_output_1_loss: 4.3269 - val_yolo_output_2_loss: 4.3915 - lr: 1.0000e-04
Epoch 8/50
715/715 [==============================] - ETA: 0s - loss: 17.1593 - yolo_output_0_loss: 0.9577 - yolo_output_1_loss: 1.7566 - yolo_output_2_loss: 3.8395
Epoch 00008: saving model to checkpoints/yolov3_train_8.tf
715/715 [==============================] - 290s 405ms/step - loss: 17.1593 - yolo_output_0_loss: 0.9577 - yolo_output_1_loss: 1.7566 - yolo_output_2_loss: 3.8395 - val_loss: 23.6398 - val_yolo_output_0_loss: 5.1509 - val_yolo_output_1_loss: 4.4299 - val_yolo_output_2_loss: 3.4624 - lr: 1.0000e-04
Epoch 9/50
715/715 [==============================] - ETA: 0s - loss: 15.6915 - yolo_output_0_loss: 0.8240 - yolo_output_1_loss: 1.3136 - yolo_output_2_loss: 2.9669
Epoch 00009: saving model to checkpoints/yolov3_train_9.tf
715/715 [==============================] - 289s 404ms/step - loss: 15.6915 - yolo_output_0_loss: 0.8240 - yolo_output_1_loss: 1.3136 - yolo_output_2_loss: 2.9669 - val_loss: 23.4546 - val_yolo_output_0_loss: 5.0880 - val_yolo_output_1_loss: 4.7778 - val_yolo_output_2_loss: 3.0118 - lr: 1.0000e-04
Epoch 10/50
715/715 [==============================] - ETA: 0s - loss: 14.5766 - yolo_output_0_loss: 0.7202 - yolo_output_1_loss: 1.0361 - yolo_output_2_loss: 2.2553
Epoch 00010: saving model to checkpoints/yolov3_train_10.tf
715/715 [==============================] - 290s 406ms/step - loss: 14.5766 - yolo_output_0_loss: 0.7202 - yolo_output_1_loss: 1.0361 - yolo_output_2_loss: 2.2553 - val_loss: 23.7491 - val_yolo_output_0_loss: 5.8852 - val_yolo_output_1_loss: 4.5641 - val_yolo_output_2_loss: 2.7482 - lr: 1.0000e-04
Epoch 11/50
715/715 [==============================] - ETA: 0s - loss: 13.7487 - yolo_output_0_loss: 0.6390 - yolo_output_1_loss: 0.8311 - yolo_output_2_loss: 1.7401
Epoch 00011: saving model to checkpoints/yolov3_train_11.tf
715/715 [==============================] - 292s 408ms/step - loss: 13.7487 - yolo_output_0_loss: 0.6390 - yolo_output_1_loss: 0.8311 - yolo_output_2_loss: 1.7401 - val_loss: 23.7634 - val_yolo_output_0_loss: 6.1159 - val_yolo_output_1_loss: 4.7568 - val_yolo_output_2_loss: 2.3672 - lr: 1.0000e-04
Epoch 12/50
715/715 [==============================] - ETA: 0s - loss: 13.1561 - yolo_output_0_loss: 0.5972 - yolo_output_1_loss: 0.6994 - yolo_output_2_loss: 1.3506
Epoch 00012: saving model to checkpoints/yolov3_train_12.tf
715/715 [==============================] - 290s 405ms/step - loss: 13.1561 - yolo_output_0_loss: 0.5972 - yolo_output_1_loss: 0.6994 - yolo_output_2_loss: 1.3506 - val_loss: 23.7688 - val_yolo_output_0_loss: 5.7243 - val_yolo_output_1_loss: 5.2128 - val_yolo_output_2_loss: 2.3391 - lr: 1.0000e-04
Epoch 13/50
715/715 [==============================] - ETA: 0s - loss: 12.6940 - yolo_output_0_loss: 0.5004 - yolo_output_1_loss: 0.6218 - yolo_output_2_loss: 1.0971
Epoch 00013: saving model to checkpoints/yolov3_train_13.tf
715/715 [==============================] - 291s 407ms/step - loss: 12.6940 - yolo_output_0_loss: 0.5004 - yolo_output_1_loss: 0.6218 - yolo_output_2_loss: 1.0971 - val_loss: 23.9774 - val_yolo_output_0_loss: 5.8261 - val_yolo_output_1_loss: 5.3457 - val_yolo_output_2_loss: 2.3514 - lr: 1.0000e-04
Epoch 14/50
715/715 [==============================] - ETA: 0s - loss: 12.3385 - yolo_output_0_loss: 0.4928 - yolo_output_1_loss: 0.5467 - yolo_output_2_loss: 0.8634
Epoch 00014: saving model to checkpoints/yolov3_train_14.tf
715/715 [==============================] - 291s 407ms/step - loss: 12.3385 - yolo_output_0_loss: 0.4928 - yolo_output_1_loss: 0.5467 - yolo_output_2_loss: 0.8634 - val_loss: 24.5731 - val_yolo_output_0_loss: 6.5884 - val_yolo_output_1_loss: 5.1778 - val_yolo_output_2_loss: 2.3918 - lr: 1.0000e-04
Epoch 15/50
715/715 [==============================] - ETA: 0s - loss: 12.0355 - yolo_output_0_loss: 0.4475 - yolo_output_1_loss: 0.4991 - yolo_output_2_loss: 0.6954
Epoch 00015: saving model to checkpoints/yolov3_train_15.tf
715/715 [==============================] - 289s 404ms/step - loss: 12.0355 - yolo_output_0_loss: 0.4475 - yolo_output_1_loss: 0.4991 - yolo_output_2_loss: 0.6954 - val_loss: 24.6375 - val_yolo_output_0_loss: 6.0910 - val_yolo_output_1_loss: 5.6231 - val_yolo_output_2_loss: 2.5518 - lr: 1.0000e-04
Epoch 16/50
715/715 [==============================] - ETA: 0s - loss: 11.8657 - yolo_output_0_loss: 0.4290 - yolo_output_1_loss: 0.4797 - yolo_output_2_loss: 0.6056
Epoch 00016: saving model to checkpoints/yolov3_train_16.tf
715/715 [==============================] - 292s 409ms/step - loss: 11.8657 - yolo_output_0_loss: 0.4290 - yolo_output_1_loss: 0.4797 - yolo_output_2_loss: 0.6056 - val_loss: 25.8418 - val_yolo_output_0_loss: 6.4979 - val_yolo_output_1_loss: 6.4148 - val_yolo_output_2_loss: 2.5979 - lr: 1.0000e-04
Epoch 17/50
715/715 [==============================] - ETA: 0s - loss: 11.6607 - yolo_output_0_loss: 0.4017 - yolo_output_1_loss: 0.4542 - yolo_output_2_loss: 0.4942
Epoch 00017: saving model to checkpoints/yolov3_train_17.tf
715/715 [==============================] - 292s 409ms/step - loss: 11.6607 - yolo_output_0_loss: 0.4017 - yolo_output_1_loss: 0.4542 - yolo_output_2_loss: 0.4942 - val_loss: 25.9489 - val_yolo_output_0_loss: 6.7254 - val_yolo_output_1_loss: 6.2932 - val_yolo_output_2_loss: 2.6406 - lr: 1.0000e-04
Epoch 18/50
715/715 [==============================] - ETA: 0s - loss: 11.4946 - yolo_output_0_loss: 0.3758 - yolo_output_1_loss: 0.4128 - yolo_output_2_loss: 0.4374
Epoch 00018: saving model to checkpoints/yolov3_train_18.tf
715/715 [==============================] - 286s 400ms/step - loss: 11.4946 - yolo_output_0_loss: 0.3758 - yolo_output_1_loss: 0.4128 - yolo_output_2_loss: 0.4374 - val_loss: 25.9633 - val_yolo_output_0_loss: 6.7053 - val_yolo_output_1_loss: 6.2931 - val_yolo_output_2_loss: 2.7149 - lr: 1.0000e-04
Epoch 19/50
715/715 [==============================] - ETA: 0s - loss: 11.3196 - yolo_output_0_loss: 0.3771 - yolo_output_1_loss: 0.3616 - yolo_output_2_loss: 0.3506
Epoch 00019: ReduceLROnPlateau reducing learning rate to 9.999999747378752e-06.

Epoch 00019: saving model to checkpoints/yolov3_train_19.tf
715/715 [==============================] - 290s 405ms/step - loss: 11.3196 - yolo_output_0_loss: 0.3771 - yolo_output_1_loss: 0.3616 - yolo_output_2_loss: 0.3506 - val_loss: 26.3280 - val_yolo_output_0_loss: 6.9310 - val_yolo_output_1_loss: 6.3072 - val_yolo_output_2_loss: 2.8805 - lr: 1.0000e-04
Epoch 20/50
715/715 [==============================] - ETA: 0s - loss: 10.8853 - yolo_output_0_loss: 0.2090 - yolo_output_1_loss: 0.2390 - yolo_output_2_loss: 0.2333
Epoch 00020: saving model to checkpoints/yolov3_train_20.tf
715/715 [==============================] - 291s 408ms/step - loss: 10.8853 - yolo_output_0_loss: 0.2090 - yolo_output_1_loss: 0.2390 - yolo_output_2_loss: 0.2333 - val_loss: 26.0611 - val_yolo_output_0_loss: 6.5314 - val_yolo_output_1_loss: 6.4562 - val_yolo_output_2_loss: 2.8753 - lr: 1.0000e-05
Epoch 21/50
715/715 [==============================] - ETA: 0s - loss: 10.6879 - yolo_output_0_loss: 0.1426 - yolo_output_1_loss: 0.1790 - yolo_output_2_loss: 0.1749
Epoch 00021: saving model to checkpoints/yolov3_train_21.tf
715/715 [==============================] - 288s 403ms/step - loss: 10.6879 - yolo_output_0_loss: 0.1426 - yolo_output_1_loss: 0.1790 - yolo_output_2_loss: 0.1749 - val_loss: 26.3808 - val_yolo_output_0_loss: 6.6464 - val_yolo_output_1_loss: 6.6074 - val_yolo_output_2_loss: 2.9432 - lr: 1.0000e-05
Epoch 22/50
715/715 [==============================] - ETA: 0s - loss: 10.6148 - yolo_output_0_loss: 0.1238 - yolo_output_1_loss: 0.1609 - yolo_output_2_loss: 0.1553
Epoch 00022: saving model to checkpoints/yolov3_train_22.tf
715/715 [==============================] - 291s 407ms/step - loss: 10.6148 - yolo_output_0_loss: 0.1238 - yolo_output_1_loss: 0.1609 - yolo_output_2_loss: 0.1553 - val_loss: 26.5839 - val_yolo_output_0_loss: 6.7420 - val_yolo_output_1_loss: 6.6507 - val_yolo_output_2_loss: 3.0263 - lr: 1.0000e-05
Epoch 23/50
715/715 [==============================] - ETA: 0s - loss: 10.5493 - yolo_output_0_loss: 0.1082 - yolo_output_1_loss: 0.1481 - yolo_output_2_loss: 0.1396
Epoch 00023: saving model to checkpoints/yolov3_train_23.tf
715/715 [==============================] - 294s 411ms/step - loss: 10.5493 - yolo_output_0_loss: 0.1082 - yolo_output_1_loss: 0.1481 - yolo_output_2_loss: 0.1396 - val_loss: 26.9190 - val_yolo_output_0_loss: 6.8917 - val_yolo_output_1_loss: 6.8101 - val_yolo_output_2_loss: 3.0763 - lr: 1.0000e-05
Epoch 24/50
715/715 [==============================] - ETA: 0s - loss: 10.4861 - yolo_output_0_loss: 0.0964 - yolo_output_1_loss: 0.1365 - yolo_output_2_loss: 0.1267
Epoch 00024: saving model to checkpoints/yolov3_train_24.tf
715/715 [==============================] - 292s 409ms/step - loss: 10.4861 - yolo_output_0_loss: 0.0964 - yolo_output_1_loss: 0.1365 - yolo_output_2_loss: 0.1267 - val_loss: 27.1788 - val_yolo_output_0_loss: 7.0300 - val_yolo_output_1_loss: 6.9188 - val_yolo_output_2_loss: 3.1187 - lr: 1.0000e-05
Epoch 25/50
715/715 [==============================] - ETA: 0s - loss: 10.4143 - yolo_output_0_loss: 0.0841 - yolo_output_1_loss: 0.1229 - yolo_output_2_loss: 0.1135
Epoch 00025: saving model to checkpoints/yolov3_train_25.tf
715/715 [==============================] - 291s 406ms/step - loss: 10.4143 - yolo_output_0_loss: 0.0841 - yolo_output_1_loss: 0.1229 - yolo_output_2_loss: 0.1135 - val_loss: 27.3874 - val_yolo_output_0_loss: 7.1310 - val_yolo_output_1_loss: 6.9929 - val_yolo_output_2_loss: 3.1886 - lr: 1.0000e-05
Epoch 26/50
715/715 [==============================] - ETA: 0s - loss: 10.3489 - yolo_output_0_loss: 0.0750 - yolo_output_1_loss: 0.1148 - yolo_output_2_loss: 0.1050
Epoch 00026: saving model to checkpoints/yolov3_train_26.tf
715/715 [==============================] - 288s 402ms/step - loss: 10.3489 - yolo_output_0_loss: 0.0750 - yolo_output_1_loss: 0.1148 - yolo_output_2_loss: 0.1050 - val_loss: 27.5807 - val_yolo_output_0_loss: 7.1429 - val_yolo_output_1_loss: 7.1559 - val_yolo_output_2_loss: 3.2496 - lr: 1.0000e-05
Epoch 27/50
715/715 [==============================] - ETA: 0s - loss: 10.2810 - yolo_output_0_loss: 0.0682 - yolo_output_1_loss: 0.1074 - yolo_output_2_loss: 0.0963
Epoch 00027: saving model to checkpoints/yolov3_train_27.tf
715/715 [==============================] - 290s 405ms/step - loss: 10.2810 - yolo_output_0_loss: 0.0682 - yolo_output_1_loss: 0.1074 - yolo_output_2_loss: 0.0963 - val_loss: 27.7404 - val_yolo_output_0_loss: 7.2319 - val_yolo_output_1_loss: 7.1785 - val_yolo_output_2_loss: 3.3449 - lr: 1.0000e-05
Epoch 28/50
715/715 [==============================] - ETA: 0s - loss: 10.2145 - yolo_output_0_loss: 0.0643 - yolo_output_1_loss: 0.1012 - yolo_output_2_loss: 0.0884
Epoch 00028: saving model to checkpoints/yolov3_train_28.tf
715/715 [==============================] - 293s 409ms/step - loss: 10.2145 - yolo_output_0_loss: 0.0643 - yolo_output_1_loss: 0.1012 - yolo_output_2_loss: 0.0884 - val_loss: 27.9862 - val_yolo_output_0_loss: 7.4295 - val_yolo_output_1_loss: 7.2403 - val_yolo_output_2_loss: 3.3808 - lr: 1.0000e-05
Epoch 29/50
715/715 [==============================] - ETA: 0s - loss: 10.1470 - yolo_output_0_loss: 0.0596 - yolo_output_1_loss: 0.0967 - yolo_output_2_loss: 0.0807
Epoch 00029: ReduceLROnPlateau reducing learning rate to 9.999999747378752e-07.

Epoch 00029: saving model to checkpoints/yolov3_train_29.tf
715/715 [==============================] - 293s 410ms/step - loss: 10.1470 - yolo_output_0_loss: 0.0596 - yolo_output_1_loss: 0.0967 - yolo_output_2_loss: 0.0807 - val_loss: 28.1393 - val_yolo_output_0_loss: 7.3661 - val_yolo_output_1_loss: 7.4419 - val_yolo_output_2_loss: 3.4473 - lr: 1.0000e-05
Epoch 00029: early stopping
```

Reply:
Ah mb, this is for object not classification

Reply:
I believe someone can try to use binary_crossentropy and see if it works.
If not, I will revise it back to sparse_categorical_crossentropy.

Reply:
Original YoloV3 also uses binary_crossentropy, so it should work fine with one_hot.

Reply:
https://blog.csdn.net/leviopku/article/details/82660381
https://github.com/qqwweee/keras-yolo3/blob/master/yolo3/model.py
It seems that binary_crossentropy is necessary.

Reply:
![image](https://user-images.githubusercontent.com/38276635/92070824-ad58f680-eddf-11ea-9201-8433f85e494e.png)